aten/src/ATen/core/Vitals.cpp

```diff
@@ -30,5 +33,26 @@ bool torchVitalEnabled() {
   return enabled;
 }
 
+bool APIVitals::setVital(
+    const std::string& vital_name,
```

Comment:
Curious why this is a string and not an enum.

Reply:
Adopting Bram's pattern. The idea is that people can adhoc declare these, I suppose. I wouldn't be opposed to an enum here.